src/applications/find-va-forms/containers/SearchResults.jsx

```diff
@@ -1,92 +1,156 @@
-      </h2>
-    );
-  }
-
-  return (
-    <>
-      <h2 className="vads-u-font-size--lg vads-u-margin-top--1p5 vads-u-font-weight--normal">
-        Showing results for "<strong>{query}</strong>"
-      </h2>
-
-      <SortableTable
-        className="vads-u-margin--0"
-        currentSort={{ order: 'ASC', value: 'title' }}
-        data={results}
-        fields={tableFields}
-      />
-    </>
-  );
-};
-
-SearchResults.propTypes = {
-  // From mapStateToProps.
-  fetching: PropTypes.bool.isRequired,
-  query: PropTypes.string.isRequired,
-  results: PropTypes.arrayOf(
-    PropTypes.shape({
-      id: PropTypes.string.isRequired,
-      type: PropTypes.string.isRequired,
-      attributes: PropTypes.shape({
+class SearchResults extends Component {
+  static propTypes = {
+    // From mapStateToProps.
+    fetching: PropTypes.bool.isRequired,
+    query: PropTypes.string.isRequired,
+    results: PropTypes.arrayOf(
+      PropTypes.shape({
+        // Original form data key-value pairs.
         firstIssuedOn: PropTypes.string.isRequired,
         formName: PropTypes.string.isRequired,
+        id: PropTypes.string.isRequired,
         lastRevisionOn: PropTypes.string.isRequired,
         pages: PropTypes.number.isRequired,
         sha256: PropTypes.string.isRequired,
         title: PropTypes.string.isRequired,
+        type: PropTypes.string.isRequired,
         url: PropTypes.string.isRequired,
+        // Table field labels that can be JSX.
+        idLabel: PropTypes.node.isRequired,
+        titleLabel: PropTypes.node.isRequired,
+        descriptionLabel: PropTypes.node.isRequired,
+        availableOnlineLabel: PropTypes.node.isRequired,
       }).isRequired,
-    }).isRequired,
-  ),
-};
+    ),
+    // From mapDispatchToProps.
+    updateResults: PropTypes.func.isRequired,
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      selectedFieldLabel: 'idLabel',
+      selectedFieldOrder: 'ASC',
+    };
+  }
+
+  onHeaderClick = (fieldLabel, order) => {
+    const { selectedFieldLabel, selectedFieldOrder } = this.state;
+
+    // Derive the opposite sort order.
+    const oppositeFieldOrder = selectedFieldOrder === 'ASC' ? 'DESC' : 'ASC';
+
+    // Sort the results.
+    this.sortResults(
+      fieldLabel,
+      fieldLabel === selectedFieldLabel ? oppositeFieldOrder : order,
+    );
+  };
+
+  sortResults = (fieldLabel, selectedFieldOrder = 'ASC') => {
+    const { results, updateResults } = this.props;
+
+    // Update local state for SortableTable.
+    this.setState({ selectedFieldLabel: fieldLabel, selectedFieldOrder });
+
+    // Derive the original field (not the JSX label).
+    const field = fieldLabel.replace('Label', '');
+
+    // Derive the sorted results.
+    const sortedResults = orderBy(
+      results,
+      field,
+      selectedFieldOrder.toLowerCase(),
+    );
+
+    // Sort the results and update them in our store.
+    updateResults(sortedResults);
```

Comment:
This is kind of a big suggestion, but it feels a little weird having the sort rules (field and asc/desc) in the component but the data in Redux, requiring us to `setState` and then `dispatch`. It might be a little cleaner to instead dispatch a sort action from this component that the reducer then responds to by sorting the data structure in state.

Reply:
I was thinking about moving both of these key-value pairs from `<SearchResults />` state to Redux:
```
selectedFieldLabel
selectedFieldOrder
```

I'll go ahead and make this change! :)

Reply:
Oh! I just reread your comment and I think there's a misunderstanding of what `selectedFieldLabel` and `selectedFieldOrder` are used for:

`<SortableTable />`takes in a `currentSort={{ order: selectedFieldOrder, value: selectedFieldLabel }}` prop, which is used to display an up-arrow and down-arrow on the column header depending on how it's sorted.